### PR TITLE
Pollution db

### PR DESCRIPTION
**Pull request by gusteran**
- Add datastore functionality for: Water Pollution at the town level, Water Systems, Contaminants, and Violation instances
- Can pull data from datastore (much faster than scraping from EPA) once stored
- Use Logger to log info and error messages
- Use Object(Input/Output)Streams to serialize and deserialize data for datastore

### CODE REVIEW

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 1]
```diff
@@ -134,23 +225,73 @@ public WaterContaminant(CSVRecord csvRecord) {
...
             violations = new HashMap<String, ArrayList<String>>();
         }
-    
+
+        public WaterContaminant(Entity violationEntity) throws IOException {
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            contaminantCode = ((Long) violationEntity.getProperty(CCODE_PROPERTY)).intValue();
+            Key contamKey = KeyFactory.createKey(CONTAMINANT_ENTITY, contaminantCode);
+            Entity contamEntity;
+			try {
```
Comment (natalie-abreu) [inline src/main/java/com/google/sps/data/WaterSystem.java:234]: nit: weird formatting

File: src/main/java/com/google/sps/servlets/WaterPollutionServlet.java [hunk 2]
```diff
@@ -100,4 +154,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         return waterSystems;
     }
 
+    public Key addToDatabase(Key key, ArrayList<WaterSystem> systems) throws IOException {
+        // DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (natalie-abreu) [inline src/main/java/com/google/sps/servlets/WaterPollutionServlet.java:158]: nit: remove commented line

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 3]
```diff
@@ -22,15 +42,26 @@
     public static final String SPLITERATOR = "\",\"";
     public static final int TOTAL_CELL_COUNT = 21;
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/data/WaterSystem.java:43]: What is this magic number 21?

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 4]
```diff
@@ -22,15 +42,26 @@
...
 
+    public static final String WATER_SYSTEM_ENTITY = "WaterSystem";
+    public static final String PWSID_PROPERTY = "pwsid";
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/data/WaterSystem.java:46]: What is pwsid?

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 5]
```diff
@@ -22,15 +42,26 @@
...
     private String name;
     private ArrayList<WaterContaminant> contaminants;
     private String state;
     private String city;
     private String county;
-    private double populationServed;
+    private int populationServed;
 
-    public WaterSystem(String pwsid, String name, String state, String city, String county, double populationServed) {
+    // DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/data/WaterSystem.java:62]: Remove the commented out line?

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 6]
```diff
@@ -59,14 +90,14 @@ public WaterSystem(String pwsid) {
                 this.county = cells[46];
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/data/WaterSystem.java:90]: This has potential to cause a lot of bugs in future. Also it is not clean. Move these number to static const?

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 7]
```diff
@@ -59,14 +90,14 @@ public WaterSystem(String pwsid) {
...
+                logger.error("There is no data for " + pwsid);
             }
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
-    public WaterSystem(CSVRecord csvRecord){
+    public WaterSystem(CSVRecord csvRecord) {
         this.pwsid = csvRecord.get(0);
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/data/WaterSystem.java:101]: Same as above

File: src/main/java/com/google/sps/servlets/WaterPollutionServlet.java [hunk 8]
```diff
@@ -59,21 +71,57 @@
...
     @Override
     public void init() {
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         response.setContentType("application/json");
+        Entity entity;
+        try {
+			entity = datastore.get(getKeyFromLocality(request.getParameter(TOWN_PARAMATER), request.getParameter(STATE_PARAMATER)));
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/servlets/WaterPollutionServlet.java:85]: Formatting error here?

Comment (gusteran) [reply src/main/java/com/google/sps/servlets/WaterPollutionServlet.java:85]: That's strange, it doesn't look like that on my editor

**Pull request merged**